instrumentation/opentelemetry-instrumentation-asyncpg/tests/test_asyncpg_wrapper.py

```diff
@@ -34,3 +37,32 @@ def test_duplicated_uninstrumentation(self):
             self.assertFalse(
                 hasattr(method, "_opentelemetry_ext_asyncpg_applied")
             )
+
+    def test_cursor_instrumentation(self):
+        def assert_wrapped(assert_fnc):
+            for cls, methods in [
+                (cursor.Cursor, ("forward", "fetch", "fetchrow")), 
+                (cursor.CursorIterator, ("__anext__", ))
+            ]:
+                for method_name in methods:
+                    method = getattr(cls, method_name, None)
+                    assert_fnc(isinstance(method, ObjectProxy), f"{method} isinstance {type(method)}")
+
+
+        assert_wrapped(self.assertFalse)
+        AsyncPGInstrumentor().instrument()
+        assert_wrapped(self.assertTrue)
+        AsyncPGInstrumentor().uninstrument()
+        assert_wrapped(self.assertFalse)
+
+    async def test_cursor_span_creation(self):
+        def mock_fn(*args, **kwargs):
+            pass
+
+        cursor_mock = mock.Mock()
+
+        apg = AsyncPGInstrumentor()
+        await apg._do_cursor_execute(mock_fn, cursor_mock, "SELECT * FROM test", {})
```

Comment:
If you look at https://github.com/open-telemetry/opentelemetry-python-contrib/blob/d0500c2f8a2160ef21ae4ef3fa3f522f3eea7f94/instrumentation/opentelemetry-instrumentation-psycopg/tests/test_psycopg_integration.py#L224
You'll see:
- they are not awaiting the async function but running the async function in asyncio.run
- they are not calling the internal helpers directly but testing the instrumented function

Reply:
I saw that they use asyncio.run however they marked their function as aysnc and pytest executes it as such. This is not the case for asyncpg tests and I really do not know why. But I can use asyncio.run it just makes the code at lot less clear to read.

In our code base we have 1000'ends of async tests but there we use plain pytest style and not the unittest base classes. Something is off here and I do not know what.

About the testing itself, the only way to test it is by mocking out connection and cursor and then instrument it. It looks like I am testing the instrumented function in that case, however it is exactly the same as what I am doing. Here it is at least explicit. I will add a comment to explain that reasoning.

Reply:
> About the testing itself, the only way to test it is by mocking out connection and cursor and then instrument it. It looks like I am testing the instrumented function in that case, however it is exactly the same as what I am doing. Here it is at least explicit. I will add a comment to explain that reasoning.

If you call the instrumented function we may change the instrumentation implementation without need to touch tests